pyomo/repn/plugins/nl_writer.py

```diff
@@ -1671,17 +1696,29 @@ def handle_product_node(visitor, node, arg1, arg2):
         mult = arg1[1]
         if not mult:
             # simplify multiplication by 0 (if arg2 is zero, the
-            # simplification happens implicitly when we evaluate the
-            # constant below)
+            # simplification happens when we evaluate the constant
+            # below).  Note that this is not IEEE-754 compliant, and
+            # will map 0*inf and 0*nan to 0 (and not to nan).  We are
+            # including this for backwards compatibility with the NLv1
+            # writer, but arguably we should deprecate/remove this
+            # "feature" in the future.
```

Comment:
Any reason for us to not add a deprecation warning now?

Reply:
Two reasons: 1) it would "break" IDAES, and 2) I am still not entirely convinced as to what the correct path forward is, so I am deferring to backwards compatibility until we convince ourselves that we want to make the change.